infrastructure/indexer-update.sh

```diff
@@ -22,6 +22,16 @@ set -o pipefail # Check all commands in a pipeline
 # Update Rust (make sure we have the latest version).
 # We need rust nightly to use the save-analysis, and firefox requires recent
 # versions of Rust.
+# Before we do the update, we remove some components that we don't need and
+# that are sometimes missing. If they are missing, `rustup update` will try
+# to use a previous nightly instead that does have the components, which means
+# we end up with a slightly older rustc. Using rustc from a few days ago is
+# usualy fine, but in cases where we hit ICEs that have been fixed upstream,
+# we want the very latest rustc to get the fix. Removing these components also
+# reduces download time during `rustup update`.
+rustup component remove clippy
+rustup component remove rustfmt
```

Comment:
Hmm, we probably want `rustfmt`. Without it Gecko's bindgen-generated files would be all on one line.

Though maybe this is not an issue for the builds on automation, so if it makes your life easier go ahead.

Reply:
Yeah I don't think we use `rustfmt` in the indexer, the formatting must happen in the taskcluster build.